runtime/dancebox/tests/integration_test.rs

```diff
@@ -3598,6 +3590,277 @@ fn test_migration_holds() {
+            assert_ok!(Session::set_keys(
+                origin_of(DAVE.into()),
+                dancebox_runtime::SessionKeys {
+                    nimbus: dave_account_id,
+                },
+                vec![]
+            ));
+
+            // We make delegations to DAVE so that she is an elligible candidate.
+
+            let stake = 10 * MinimumSelfDelegation::get();
+
+            assert_ok!(PooledStaking::request_delegate(
+                origin_of(DAVE.into()),
+                DAVE.into(),
+                TargetPool::ManualRewards,
+                stake,
+            ));
+            assert_ok!(PooledStaking::request_delegate(
+                origin_of(BOB.into()),
+                DAVE.into(),
+                TargetPool::AutoCompounding,
+                stake,
+            ));
+
+            // wait few sessions for the request to be executable
+            run_to_session(3u32);
+            assert_ok!(PooledStaking::execute_pending_operations(
+                origin_of(ALICE.into()),
+                vec![
+                    PendingOperationQuery {
+                        delegator: DAVE.into(),
+                        operation: PendingOperationKey::JoiningManualRewards {
+                            candidate: DAVE.into(),
+                            at: 0
+                        }
+                    },
+                    PendingOperationQuery {
+                        delegator: BOB.into(),
+                        operation: PendingOperationKey::JoiningAutoCompounding {
+                            candidate: DAVE.into(),
+                            at: 0
+                        }
+                    }
+                ]
+            ));
+
+            // wait for next session so that DAVE is elected
+            run_to_session(4u32);
+
+            assert_eq!(
+                Session::validators(),
+                vec![AccountId::from(ALICE), AccountId::from(DAVE)]
+            );
+
+            let account: AccountId = DAVE.into();
+            let balance_before = System::account(account.clone()).data.free;
+            let summary = (0..100)
+                .find_map(|_| {
+                    let summary = run_block();
+                    if summary.author_id == DAVE.into() {
+                        Some(summary)
+                    } else {
+                        None
+                    }
+                })
+                .unwrap_or_else(|| panic!("DAVE doesn't seem to author any blocks"));
+            let balance_after = System::account(account).data.free;
+
+            let all_rewards = RewardsPortion::get() * summary.inflation;
+            // rewards are shared between orchestrator and registered paras
+            let orchestrator_rewards = all_rewards / 3;
+            let candidate_rewards = orchestrator_rewards * 2 / 10;
+
+            assert_eq!(
+                candidate_rewards + 1, // TODO: account for rounding properly :p
```

Comment:
Are we going to fix this? Maybe add another test that checks that the rewarded container chains plus the orchestrator is exactly equal to all_rewards?

Reply:
I have a bunch of those as well, probably worth creating an issue